src/geff/networkx/io.py

```diff
@@ -40,30 +41,39 @@ def get_roi(graph: nx.Graph, position_attr: str) -> tuple[tuple[float, ...], tup
     return tuple(_min.tolist()), tuple(_max.tolist())  # type: ignore
 
 
-def get_node_attrs(graph: nx.Graph) -> list[str]:
-    """Get the attribute keys present on any node in the networkx graph. Does not imply
-    that the attributes are present on all nodes.
+class NXWriterHelper(BaseWriterHelper):
+    """
+    Helper class for writing node or edge attributes to a zarr group.
 
     Args:
-        graph (nx.Graph): A networkx graph
-
-    Returns:
-        list[str]: A list of all unique node attribute keys
+        group: The zarr group to write the attribute to.
+        data_sequence: a sequence of (id, data) pairs. For example, graph.nodes(data=True)
+        position_attr: The name of the position attribute. If None, no position attribute
     """
-    return list({k for n in graph.nodes for k in graph.nodes[n]})
-
 
-def get_edge_attrs(graph: nx.Graph) -> list[str]:
-    """Get the attribute keys present on any edge in the networkx graph. Does not imply
-    that the attributes are present on all edges.
+    def __init__(
+        self,
+        group: zarr.Group,
```

Comment:
one thing to consider (for slightly increased generality) would be to not take the group as an input to the BaseWriterHelper ... but rather have the consumer of the writer pass it as an argument to write_attrs?

so the class just worries about how to gather the right information... but not where it's going

Reply:
I agree. Let's put the destination in write attrs

Reply:
I moved, and it got much simpler. I don't even need to create a new class for networkx.